cmd/nerdctl/container_run_soci_linux_test.go

```diff
@@ -27,60 +26,52 @@ import (
 
 func TestRunSoci(t *testing.T) {
 	testutil.DockerIncompatible(t)
-	base := testutil.NewBase(t)
-	requiresSoci(base)
-
-	//counting initial snapshot mounts
-	initialMounts, err := exec.Command("mount").Output()
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	remoteSnapshotsInitialCount := strings.Count(string(initialMounts), "fuse.rawBridge")
-
-	if remoteSnapshotsInitialCount != 0 {
-		t.Fatalf("initial mounts count isn't zero")
-	}
-
-	//validating `nerdctl --snapshotter=soci run` and `soci rpull` behave the same using mounts
-	runOutput := base.Cmd("--snapshotter=soci", "run", "--rm", testutil.FfmpegSociImage).Out()
-	base.T.Logf("run output: %s", runOutput)
-
-	actualMounts, err := exec.Command("mount").Output()
-	if err != nil {
-		t.Fatal(err)
+	tests := []struct {
+		name                         string
+		image                        string
+		remoteSnapshotsExpectedCount int
+	}{
+		{
+			name:                         "Run with SOCI",
+			image:                        testutil.FfmpegSociImage,
+			remoteSnapshotsExpectedCount: 11,
+		},
 	}
-	remoteSnapshotsActualCount := strings.Count(string(actualMounts), "fuse.rawBridge")
-	base.T.Logf("number of actual mounts: %v", remoteSnapshotsActualCount)
 
-	rmiOutput := base.Cmd("rmi", testutil.FfmpegSociImage).Out()
-	base.T.Logf("rmi output: %s", rmiOutput)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			base := testutil.NewBase(t)
+			requiresSoci(base)
 
-	sociExecutable, err := exec.LookPath("soci")
-	if err != nil {
-		t.Fatalf("SOCI is not installed.")
-	}
+			//counting initial snapshot mounts
+			initialMounts, err := exec.Command("mount").Output()
+			if err != nil {
+				t.Fatal(err)
+			}
 
-	rpullCmd := exec.Command(sociExecutable, []string{"image", "rpull", testutil.FfmpegSociImage}...)
+			remoteSnapshotsInitialCount := strings.Count(string(initialMounts), "fuse.rawBridge")
 
-	rpullCmd.Env = os.Environ()
+			runOutput := base.Cmd("--snapshotter=soci", "run", "--rm", testutil.FfmpegSociImage).Out()
+			base.T.Logf("run output: %s", runOutput)
 
-	err = rpullCmd.Run()
-	if err != nil {
-		t.Fatal(err)
-	}
+			actualMounts, err := exec.Command("mount").Output()
+			if err != nil {
+				t.Fatal(err)
+			}
+			remoteSnapshotsActualCount := strings.Count(string(actualMounts), "fuse.rawBridge")
+			base.T.Logf("number of actual mounts: %v", remoteSnapshotsActualCount)
 
-	expectedMounts, err := exec.Command("mount").Output()
-	if err != nil {
-		t.Fatal(err)
-	}
+			rmiOutput := base.Cmd("rmi", testutil.FfmpegSociImage).Out()
+			base.T.Logf("rmi output: %s", rmiOutput)
 
-	remoteSnapshotsExpectedCount := strings.Count(string(expectedMounts), "fuse.rawBridge")
-	base.T.Logf("number of expected mounts: %v", remoteSnapshotsExpectedCount)
+			remoteSnapshotsExpectedCount := 11
```

Comment:
this should be `tt.remoteSnapshotsExpectedCount` right?

(also since `tt` is used within `t.Run`, it will be safer to add a line `tt := tt` between line 41 and 42)

Reply:
Oh, sorry. I initially did not make a struct for this and hardcoded in the values instead. Fixed this.

> (also since tt is used within t.Run, it will be safer to add a line tt := tt between line 41 and 42)

From my understanding of this Go semantics intricacy [here](https://github.com/golang/go/discussions/56010), it seems this is only a problem when making references to any of the loop variables outside of the context of this loop. Given this, I think it would be better to not have an extra line of code. But, if I'm wrong, or if you feel particularly strongly about this, I'm happy to add that line.

Reply:
> this is only a problem when making references to any of the loop variables outside of the context of this loop

My understanding is it will be a problem  if any goroutines created within the for-loop body use the loop variables. For example, if we parallel these `t.Run` sub tests, it will be a problem. Here since you're not parallel these sub tests, it should be okay.